soc/arm64/arm/fvp_aemv8r/arm_mpu_regions.c

```diff
@@ -33,7 +33,11 @@ static const struct arm_mpu_region mpu_regions[] = {
 
 	/* Region 3 zephyr data */
 	MPU_REGION_ENTRY("SRAM_2",
+#ifdef CONFIG_USERSPACE
+			 (uintptr_t)_app_smem_start,
+#else
 			 (uintptr_t)__kernel_ram_start,
+#endif
```

Comment:
`_app_smem_start` is only for symbols in memory partitions, which should have different permissions than `.data`.

Reply:
This is only for the init stage, and both `_app_smem_start` and `.data` are inited as `P_RW_U_NA`. These sections act as background/default regions (actually they will be marked as `dyn_region` after MPU being inited), and then they will be reconfigured during the thread init and context-switching dynamically.

Reply:
Just for clarification, will `.data` have its own permissions or will it be grouped together with shared member?

Reply:
The `.data` and `_app_smem_start` will be reconfiged as their own separated permissions. I remember there are some test cases that will test this permission.

Reply:
@dcpleung Would you like to have a look again?:)